docker/build-tools/Dockerfile

```diff
@@ -432,3 +430,89 @@ RUN chmod 777 /go && \
 WORKDIR /
 
 ENTRYPOINT ["/usr/local/bin/docker-entrypoint"]
+
+##############
+# Clang+LLVM
+##############
+
+FROM ubuntu:bionic as clang_context
+
+# hadolint ignore=DL3008
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    xz-utils \
+    wget \
+    ca-certificates \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
+
+ENV LLVM_VERSION=9.0.0
+ENV LLVM_TARGET=x86_64-linux-gnu-ubuntu-16.04
+
+ENV LLVM_BASE_URL=http://releases.llvm.org/${LLVM_VERSION}
+ENV LLVM_ARCHIVE=clang+llvm-${LLVM_VERSION}-${LLVM_TARGET}
+ENV LLVM_URL=${LLVM_BASE_URL}/${LLVM_ARCHIVE}.tar.xz
+ENV LLVM_DIRECTORY=/usr/lib/llvm
+
+RUN wget ${LLVM_URL}
+RUN tar -xJf ${LLVM_ARCHIVE}.tar.xz -C /tmp
+RUN mkdir -p ${LLVM_DIRECTORY}
+RUN mv /tmp/${LLVM_ARCHIVE}/* ${LLVM_DIRECTORY}/
```

Comment:
Could you incorporate changes from https://github.com/istio/test-infra/pull/2042?

Reply:
I think we might not need that change. This docker file uses multi stage docker builds. The artifacts from intermediate steps will not be included in the final image, since we only copy what we want from these intermediate images. wdyt?

Reply:
And do we need clang and llvm with RBE? Should we only install bazel?

Reply:
> I think we might not need that change. This docker file uses multi stage docker builds. The artifacts from intermediate steps will not be included in the final image, since we only copy what we want from these intermediate images. wdyt?

I see that you download the tarball and never remove it. I'm not sure how multi-stage docker builds work, but I don't see how it could automatically figure out that you don't want the tarball.

Anyway, since this was merged already, is the tarball there in the resulting image?

> And do we need clang and llvm with RBE? Should we only install bazel?

Not everything is using RBE yet (e.g. GKE builds, postsubmit in OSS).

Reply:
> Anyway, since this was merged already, is the tarball there in the resulting image?

No it is not there. In the last step to build the final image, it only copies the necessary library and binaries:
https://github.com/istio/tools/blob/ff233f25bab3247b7ba2429313e7a470b978fda3/docker/build-tools/Dockerfile#L507

Reply:
Great, thanks!